src/types/module.rs

```diff
@@ -194,11 +195,50 @@ impl PyModule {
     /// ```rust,ignore
     /// m.add("also_double", wrap_pyfunction!(double)(py));
     /// ```
-    pub fn add_wrapped(&self, wrapper: &impl Fn(Python) -> PyObject) -> PyResult<()> {
-        let function = wrapper(self.py());
-        let name = function
-            .getattr(self.py(), "__name__")
-            .expect("A function or module must have a __name__");
-        self.add(name.extract(self.py()).unwrap(), function)
+    ///
+    /// **This function will be deprecated in the next release. Please use the specific
+    /// [add_function] and [add_module] functions instead.**
+    pub fn add_wrapped<'a, T>(&'a self, wrapper: &impl Fn(Python<'a>) -> T) -> PyResult<()>
+    where
+        T: IntoPyCallbackOutput<PyObject>,
+    {
+        let function = wrapper(self.py()).convert(self.py())?;
+        let name = function.getattr(self.py(), "__name__")?;
+        self.add(name.extract(self.py())?, function)
+    }
+
+    /// Adds a (sub)module to a module.
```

Comment:
I think it's fine to just call it a submodule. (I see you got this wording from add_wrapped)

```suggestion
    /// Add a submodule to this module.
```